src/python-frontend/astgen.py

```diff
@@ -0,0 +1,23 @@
+import ast
+import sys
+from ast2json import ast2json
+import json
+
+def main():
+    if len(sys.argv) != 2:
+        print("Usage: python astgen.py <file path>")
+        return
+
+    filename = sys.argv[1]
+
+    with open(filename, "r") as source:
+        tree = ast.parse(source.read())
+
+    ast_json = ast2json(tree)
+    json_filename = "/tmp/ast.json"
```

Comment:
Is there any way to make this a bit more generic? Typically the convention in `/tmp` is that existing files should not be overwritten.

Reply:
You could use the tmp_path class (in util/filesystem.h)

Reply:
Done.